src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityFissionReactor.java

```diff
@@ -0,0 +1,1104 @@
+        // Take in coolant, take in fuel, update reactor, output steam
+
+        if (!this.getWorld().isRemote && this.getOffsetTimer() % 20 == 0) {
+            if (this.lockingState == LockingState.LOCKED) {
+                // Coolant handling
+                if (this.getOffsetTimer() % 100 == 0) {
+                    if (isFlowingCorrectly) {
+                        if (getWorld().rand.nextDouble() > (1 - 0.01 * this.getNumMaintenanceProblems())) {
+                            isFlowingCorrectly = false;
+                        }
+                    } else {
+                        if (getWorld().rand.nextDouble() > 0.12 * this.getNumMaintenanceProblems()) {
+                            isFlowingCorrectly = true;
+                        }
+                    }
+                }
+
+                // Fuel handling
+                if (this.fissionReactor.isDepleted()) {
+                    boolean canWork = true;
+                    boolean needsReset = false;
+                    for (IFuelRodHandler fuelImport : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+                        if (fuelImport.getStackHandler().extractItem(0, 1, true).isEmpty()) {
+                            canWork = false;
+                            this.lockingState = LockingState.MISSING_FUEL;
+                            break;
+                        } else if (!((MetaTileEntityFuelRodImportBus) fuelImport).getExportHatch(this.height - 1)
+                                .getExportItems().insertItem(0,
+                                        OreDictUnifier.get(OrePrefix.fuelRodHotDepleted, fuelImport.getFuel()), true)
+                                .isEmpty()) {
+                                    // We still need to know if the output is blocked, even if the recipe doesn't start
+                                    // yet
+                                    canWork = false;
+                                    this.lockingState = LockingState.FUEL_CLOGGED;
+                                    break;
+                                }
+                    }
+
+                    for (IFuelRodHandler fuelImport : this.getAbilities(MultiblockAbility.IMPORT_FUEL_ROD)) {
+                        if (fissionReactor.needsOutput) {
+                            ((MetaTileEntityFuelRodImportBus) fuelImport).getExportHatch(this.height - 1)
+                                    .getExportItems().insertItem(0,
+                                            OreDictUnifier.get(OrePrefix.fuelRodHotDepleted,
+                                                    fuelImport.getPartialFuel()),
+                                            false);
+                            this.fissionReactor.fuelMass -= 60;
+                        }
+                        if (canWork) {
+                            fuelImport.getStackHandler().extractItem(0, 1, false);
+                            needsReset |= fuelImport.setPartialFuel(fuelImport.getFuel());
```

Comment:
Note that this does not short-circuit. It will always run the right side of the operator. Is that intentional?